Merged setup with learn course and v0

ogiles1999 opened the pull request:
**Combined the setup with new week one of data engineering**

Main changes:

- Added postgres to run a local postgres server in week 1
- Added minikube to launch local cluster
- Added kubectl to interact with k8s clusters
- Added in dbeaver section to interacts with dbs

ogiles1999 commented:
Still to do based on decision on inclusion

- heroku cli
- pgadmin

ogiles1999 commented:
> Still to do based on decision on inclusion
> 
>     * heroku cli
> 
>     * pgadmin

decided not to include

ogiles1999 commented:
Requires https://github.com/lewagon/dotfiles/pull/127

gmanchon commented:
@lucasgday can you have a look at the spanish partials modified by this PR for the data engineering setup please ?

gmanchon commented:
I had trouble setting `zsh` as the default _terminal_ in **VSCode** using the instructions, here is what I did :
![Screenshot 2022-10-26 at 19 31 16](https://user-images.githubusercontent.com/34584935/198096534-043ccca0-c659-40cf-92cf-eb057bac3458.png)
![Screenshot 2022-10-26 at 19 31 26](https://user-images.githubusercontent.com/34584935/198096532-e09912aa-0b3d-470c-a9a6-9030f2bbd35e.png)
![Screenshot 2022-10-26 at 19 31 35](https://user-images.githubusercontent.com/34584935/198096530-afa52615-0abb-49cc-a245-dacf8b05c1c6.png)

gmanchon commented:
I had trouble installing `kubectl` from the instructions. I followed the instructions from [kubernetes.io](https://kubernetes.io/docs/tasks/tools/install-kubectl-linux/) :

```
curl -LO "https://dl.k8s.io/release/$(curl -L -s https://dl.k8s.io/release/stable.txt)/bin/linux/amd64/kubectl"
curl -LO "https://dl.k8s.io/$(curl -L -s https://dl.k8s.io/release/stable.txt)/bin/linux/amd64/kubectl.sha256"
echo "$(cat kubectl.sha256)  kubectl" | sha256sum --check
sudo install -o root -g root -m 0755 kubectl /usr/local/bin/kubectl
kubectl version --client
kubectl version --client --output=yaml
```

gmanchon commented:
I was not able to `pyenv install 3.8.14` out of the box (`no acceptable c compiler found in $path`). This was solved by :

```
sudo apt-get install build-essential
```

Then it failed again with `subprocess.CalledProcessError: Command ... [Makefile:1198: install] Error 1`, so :

```
sudo apt-get install make build-essential libssl-dev zlib1g-dev \
libbz2-dev libreadline-dev libsqlite3-dev wget curl llvm \
libncursesw5-dev xz-utils tk-dev libxml2-dev libxmlsec1-dev libffi-dev liblzma-dev
```

... Then it was ok

gmanchon commented:
👆 Other than that worked like a charm 🔥

gmanchon commented:
Option : when we edit the ssh conf of the student machine in `~/.ssh/config`, it can be nice to provide a human readable name for the VM

```
Host "Data Eng 34.76.57.184"
  HostName 34.76.57.184
  IdentityFile ~/.ssh/id_ed25519
  User gaeetanc
```

ogiles1999 commented:
@gmanchon 

Great finds!

- Changed to your instruction for kubectl
- Added all the packages to sudo-apt to allow installation of 3.8.14
- Added section about updating ssh config
- Updated to 22.04 lts image

For stopping the vm the first challenge includes this so should be all good but we can add as well if it would be good to have in multiple places!

gmanchon commented on _partials/vscode_remote_ssh.md:
````diff
@@ -46,4 +46,22 @@ That's the only extension you should install on your _local_ machine, we will in
+  IdentityFile <file path for your ssh key>
+  User <username>
+```
+You can now change Host to whatever you would like to see as the name of your connection or in terminal with `ssh <Host>`!
````
``` suggestion
You can now change Host to whatever you would like to see as the name of your connection or in terminal with `ssh <Host>`! (surround the Host with double quotes if the name contains spaces `Host "my host"`)
```

brunolajoie commented:
So cool! Thanks gaetan for reviewing! If it worked for you, then I think we can merge! No big deal if we need further correction later on, 0 students are affected

brunolajoie commented on _partials/cli_tools.md:
````diff
@@ -12,7 +12,11 @@ Let's install them, along with other useful tools:
 ```bash
 sudo apt update
-sudo apt install -y vim tmux tree git ca-certificates curl jq unzip zsh apt-transport-https gnupg software-properties-common direnv sqlite3 make
+sudo apt install -y vim tmux tree git ca-certificates curl jq unzip zsh \
````
It would be amazing to break down the lines per "groups" of tools, like 
```suggestion
sudo apt install -y vim tmux tree git ca-certificates curl jq unzip zsh # basic bash navigation tools
sudo apt-install ... # python installation dependencies
sudo apt-install ... # docker & kube etc...
...
```

brunolajoie commented on _partials/ubuntu_python.md:
````diff
@@ -1,15 +1,46 @@
+
+```bash
+pip install --upgrade pip
+python -m pip install --user pipx
````
Why do we install it with `--user` ?
